target/linux/lantiq/patches-4.19/2001-NET-MIPS-lantiq-add-mdio-ioctl.patch

```diff
@@ -0,0 +1,43 @@
+--- a/drivers/net/ethernet/lantiq_xrx200.c
++++ b/drivers/net/ethernet/lantiq_xrx200.c
+@@ -1279,6 +1279,32 @@ static void xrx200_gmac_update(struct xr
+ }
+ #endif
+ 
++static int xrx200_ioctl(struct net_device *dev, struct ifreq *ifr, int cmd)
```

Comment:
with this remoteprocessor driver idea from above you may be able to drop this ioctl

Reply:
Yes, depends on how much is going to be implemented in kernel space and how much in user space. 
Do you see disadvantages in allowing these ioctls? Other ethernet drivers expose the MDIO interface as well and I was quite surprised to find out the `mdio-tool` or `phytool` do not work on the Lantiq target.

Reply:
you're patching the out-of-tree driver here. by itself this is not a problem because that's what we use in OpenWrt.
my understanding is that the upstream driver (drivers/net/ethernet/lantiq_xrx200.c) in Linux 5.4 has the same feature set as the out-of-tree driver (before your patch), so switching to the upstream driver is a matter of writing a config migration script (to switch from the old to the new driver).

I don't know if the upstream driver also supports these IOCTL already. if not we may end up in a dilemma:
- use the upstream driver and break support for the FRITZ!Box 3390 (either by accident, because nobody knows that the IOCTLs are missing in the upstream driver - or on purpose, because it affects only one device)
- keep the out-of-tree driver and live with the increased maintenance effort

a slightly off-topic but still related story from the past: In the past we switched from the out-of-tree SPI driver to the upstream one during a kernel update because I could not get the out-of-tree one working (even after spending multiple days on it).

Reply:
I don't have a problem with submitting the patch to upstream, but I suppose that more work is needed before it can be accepted, since I only implemented the bare minimum of ioctls to support the FRITZ!Box. 

Since I'm also working on the 3490: It needs the same ioctls, so more than one device would be affected.